processing/src/main/java/org/apache/carbondata/processing/store/writer/AbstractFactDataWriter.java

```diff
@@ -205,8 +205,10 @@ public AbstractFactDataWriter(CarbonFactDataHandlerModel model) {
       if (model.getNumberOfCores() > 1) {
         numberOfCores = model.getNumberOfCores() / 2;
       }
-      fallbackExecutorService = Executors.newFixedThreadPool(numberOfCores, new CarbonThreadFactory(
-          "FallbackPool:" + model.getTableName() + ", range: " + model.getBucketId()));
+      fallbackExecutorService = model.getFallBackExecutorService() != null ?
+          model.getFallBackExecutorService() :
+          Executors.newFixedThreadPool(numberOfCores, new CarbonThreadFactory(
```

Comment:
Better to use "public CarbonThreadFactory(String name, boolean withTime)", so that diff threads have diff names